taichi/program/program.cpp

```diff
@@ -455,8 +455,25 @@ std::size_t Program::get_snode_num_dynamically_allocated(SNode *snode) {
 
 Ndarray *Program::create_ndarray(const DataType type,
                                  const std::vector<int> &shape,
-                                 ExternalArrayLayout layout) {
-  ndarrays_.emplace_back(std::make_unique<Ndarray>(this, type, shape, layout));
+                                 ExternalArrayLayout layout,
+                                 bool zero_fill) {
+  auto arr = std::make_unique<Ndarray>(this, type, shape, layout);
+  if (zero_fill) {
+    Arch arch = this_thread_config().arch;
+    if (arch == Arch::x64 || arch == Arch::cuda) {
```

Comment:
or arch == Arch::arm64? I guess it will fail on M1 because arm64 uses CpuDevice as well. Maybe we can use `arch_is_cpu(arch)` here.

Reply:
good point!